db/view/view_building: replace system keyspace functions with mutation builder

Jadw1 opened the pull request:
`system.view_building_tasks` is a single partition table, so it makes more sense to use a mutation builder and generate 1 mutation per group0 command instead of generating multiple mutations.

This PR removes all `make_..._mutation()` system keyspace functions related to view building tasks and replaces them with mutation builder.

Refs https://github.com/scylladb/scylladb/issues/25929

This patch doesn't fix any bug, it only reduces number of generated mutations, no need to backport it.

Jadw1 committed:
db/view/view_building_task_mutation_builder: add helper method

Add a method to set all task's fields.

Jadw1 committed:
db/system_keyspace: replace `make_remove_view_building_task_mutation()` with mutation builder

Again, get rid of system keyspace method in favor of mutation builder,
because `system.view_building_tasks` is a single parition table.

Jadw1 commented:
@piodul ping, can we merge this?

piodul commented:
I'm not sure I agree with the motivation for the backport. Sure, if convenience were our biggest priority when backporting, then we should backport everything. However, this change is a refactor that has a chance to introduce some bugs. I would rather not backport it to potentially destabilize the 2025.4 branch.

I agree that this will cause some conflicts. However, at a glance, I don't think those conflicts will be super difficult to fix - slightly painful, yes, but nothing that requires a total rewrite of further backports.

mlitvk commented on service/migration_manager.cc:
```diff
@@ -945,27 +946,28 @@ static future<> add_view_building_tasks_mutations(storage_proxy& sp, view_ptr vi
+            builder.set_task(task);
             mlogger.trace("Creating view building task: {} with ID: {} for replica: {}", task, id, replica);
         }
+        co_return;
```
nit: now there's no reason for it to be a coroutine, wouldn't it be better to make it non-coroutine by `return make_ready_future<>()`?

Jadw1 committed:
db/system_keyspace: replace `make_view_building_task_mutation()` with mutation builder

`system.view_building_tasks` is a single partition table, so it makes
more sense to use a mutation builder and generate 1 mutation per group0
command instead of generating multiple mutations.

Jadw1 committed:
db/view/view_building_task_mutation_builder: make uuid generator optional

After scylladb/scylladb#28929 `task_uuid_generator` became necassary
dependency of `view_building_task_mutation_builder`.

However to create the generator we need `view_building_state`, which in
some parts of the code (schema_tables.cc, migration_manager.cc) requires
remote proxy to be obtained.

But sometimes we need the mutation builder to just remove some view
building task. In those cases, we don't need the uuid generator and the
remote proxy requirement is not necassary.
